docs/SOURCING-COMMITMENTS.md

```diff
@@ -0,0 +1,91 @@
+
+Also note that "adding a utxo" does **not** mean spending it! It only means adding a hash-value commitment and the pubkey, basically. Any Bitcoin utxo can be used as an effective 3-time usage token, without spending it.
+
+Be aware that Makers will see this utxo *only if* your usage is successful (they accept it and return their utxos in response), which is a reason not to use this "sourcing external" approach if you have any concerns about them getting that knowledge. Also, if you source a utxo from another non-JM wallet you will have to pass its private key in WIF compressed format to the script (it cannot be signed inside the wallet as this is not a standard Bitcoin ECDSA signature), in order to create the commitment - the usual strong warnings about handling private keys apply here, so use this approach only *very* carefully.
+
+### Commitment storage, and what if things go wrong.
+
+The commitments you've already used (just hash values) are stored in the `used` section of the file `cmttools/commitments.json` - **don't delete this file**. Although you won't need to read this file, it represents your memory of which commitments you've already used, so if you lose that record, your Taker scripts will find themselves spending a lot of time retrying commitments that the rest of the Joinmarket "network" knows are already used, and so will be rejected. This is only an inconvenience, but it could be pretty annoying. Any external commitments you sourced according to the previous section are also added here, and deleted automatically once they're used up. You can add/delete the contents of that `external` section using the previously mentioned `cmttools/add-utxo.py` script (see previous section for a brief overview of options).
+
+If in a run of `sendpayment.py` no commitment can be sourced (in tumbler it just waits), either within the internal wallet, or external as described above, a file `commitments_debug.txt` is created that will show exactly which utxos have been tried and why they failed - and gives brief instructions on what to do depending on the state of each of those. A sample file is shown at the end of this page.
+
+### Minor additional tools
+
+An extra method for `wallet-tool.py` is added: `showutxos` - this will pretty print the all the available utxos in any Joinmarket wallet. This is only potentially useful for people who run both as Maker and Taker and want to consider transferring utxos for commitments from one wallet to another (using the `-w` option to `cmttools/add-utxo.py`).
+
+Similarly a `sendtomany` function is available in `cd cmttools; python sendtomany.py --help`; read the help for details but it's as simple as it sounds, specifically creating equal sized outputs for each of the destination addresses you specify; it requires one utxo and its private key as inputs. You're prompted before broadcast, so you can check its validity.
```

Comment:
`cd scripts; python3 sendtomany.py --help`.

Reply:
@kristapsk Shouldn't it be just `python sendtomany.py --help` ? I'm assuming the user will have `jmvenv` activated here.

Reply:
Yeah, probably!